src/app/sidebar/sidenav/sidenav.tsx

```diff
@@ -0,0 +1,182 @@
+import React, { useEffect } from "react";
+import styles from "./sidenav.module.css";
+import { IconChevronDown16 } from "@dhis2/ui-icons";
+import cx from "classnames";
+
+export const Sidenav = ({ children }: React.PropsWithChildren<{}>) => (
+    <nav className={styles["sidenav-wrap"]}>{children}</nav>
+);
+
+export const SidenavItems = ({ children }: React.PropsWithChildren<{}>) => (
+    <ul className={styles["sidenav-items"]}>{children}</ul>
+);
+
+export const SidenavHeader = ({ children }) => (
+    <div className={styles["sidenav-header"]}>{children}</div>
+);
+
+export const SidenavFooter = ({ children }) => (
+    <div className={styles["sidenav-footer"]}>{children}</div>
+);
+
+interface SidenavParentProps {
+    label: string;
+    icon?: React.ReactNode;
+    forceOpen?: boolean;
+    initialOpen?: boolean;
+    disabled?: boolean;
+    children: React.ReactNode;
+}
+
+export const SidenavParent = ({
+    label,
+    icon,
+    initialOpen,
+    disabled,
+    children,
+    forceOpen,
+}: SidenavParentProps) => {
+    const [showResults, setShowResults] = React.useState(!!initialOpen);
+    // used to keep the state before forceOpen changed - so it can be reset back to previous
+    // state when forceOpen is not active anymore
+    const beforeForceOpenShowResults = React.useRef(showResults);
+
+    const toggleChildren = React.useCallback(() => {
+        setShowResults((showResults) => !showResults);
+    }, []);
+
+    React.useEffect(() => {
+        if (forceOpen != undefined) {
+            beforeForceOpenShowResults.current = showResults;
+            setShowResults(forceOpen);
+        } else {
+            setShowResults(beforeForceOpenShowResults.current);
+        }
+    }, [forceOpen]);
+
+    const isOpen = showResults;
+
```

Comment:
I actually tried something similar first, but it didn't work out due to some functionality that I think are natural to have. 

The reason for this logic is that `forceOpen` is used to open the the parent when it's filtered, and there's a match. But when the filter is cleared, we want to revert to the previous state, before it was forced open. Which would probably "just" work in your solution if we centralised the open-states in the sidebar. However that brings up my next point:

In your solution the user cannot close a parent while a filter is active, because `forceOpen` would alone decide if it should be open. This is one of the reasons for the complex logic. I guess it could be simplified, but this is what I could come up with that satisfied "my" requirements. Of course these are not strict requirements, but It's what I deemed natural for a feature like this. Maybe it's not worth the "complexity", but in essence it's just keeping a reference to the previous state. So I think this logic would need to stay, regardless if we moved the open-states to the `Sidebar`. 

I will answer the point in the main review thread here, since these are kind of related: 

> With the simplication of the forceOpen and previous state in the SidenavParent (see https://github.com/dhis2/maintenance-app-beta/pull/323#discussion_r1136409412), you could centralize the state for which parent is open in the Sidebar component by using useState that has an array with all opened labels. That way you can remove the entire HidePreventUnmount logic / component, which seems unnecessarily complicated to me. The approach of centralizing the state in a single component and than rendering only
stateless components seems to be simpler to me.

First, I think it's simpler to keep this state in the `SidebarParent`, mostly because the open state depends on the current matched route - and this is more convenient to check there, instead of having to manually `match` for each route. It's also less "wiring", since in that case the `SidebarParent` would need a callback to open the parent, which would cause re-rendering of all items. In this case the performance is negligible, but in general I don't like having a "list" of state unless necessary.
It would certainly be possible, but I don't necessarily think it will be simpler, especially since it doesn't really have any impact on the "complex" logic of this component.

Reply:
I see. I think this is something we should confirm with @cooper-joe. With the current approach, when you close a parent and then refine your search, that parent will stay closed (I guess that's OK because it's been closed on purpose).

The specs have the following two requirements:

1. Sections are automatically expanded if they contain an item(s) that matches the filter.
2. Sections that contain no items are hidden.

The first point does not mention forcing sections to be opened, just that they should be expanded when there's a filter match (which brings me back to my comment on what should happen to closed sections when changing the search input after having filtered the nav and then closed a section).
The second point mentions that sections without a match should be hidden, which is not the case right now.

Given that sections without items should be hidden, I think I'd

* always keep sections open 
* "force" the user to refine his search if there are too many sections showing (instead of making him close sections)

But as that isn't what specs say either, we need @cooper-joe's input on this topic.

Reply:
A change in the filtering input constitutes a new attempt at filtering and should "reset" the visible state of a section, in my opinion. That means, whenever a user modifies the filter input, we re-apply the logic of _Sections are automatically expanded if they contain an item(s) that matches the filter_.

Reply:
https://dhis2.atlassian.net/browse/DHIS2-15074